src/plugins/keras_support/switch_operation.cpp

```diff
@@ -0,0 +1,397 @@
+        {
+            if (!valid(i))
+            {
+                HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                    "switch_operation::eval",
+                    util::generate_error_message(
+                        "the switch primitive requires that the arguments "
+                        "given by the operands array are valid",
+                        name_, codename_));
+            }
+        }
+
+        auto this_ = this->shared_from_this();
+        return hpx::dataflow(hpx::launch::sync,
+            hpx::util::unwrapping([this_ = std::move(this_)](
+                                      primitive_arguments_type&& args)
+                                      -> primitive_argument_type {
+
+                std::array<std::size_t, PHYLANX_MAX_DIMENSIONS> then_dims =
+                    extract_numeric_value_dimensions(
+                        std::move(args[1]), this_->name_, this_->codename_);
+                std::array<std::size_t, PHYLANX_MAX_DIMENSIONS> else_dims =
+                    extract_numeric_value_dimensions(
+                        std::move(args[2]), this_->name_, this_->codename_);
+
+                if (then_dims != else_dims)
+                    HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                        "switch_operation::eval",
+                        util::generate_error_message(
+                            "the `then_expression` and `else_expression` "
+                            "should be of the same shape",
+                            this_->name_, this_->codename_));
+
+                std::array<std::size_t, PHYLANX_MAX_DIMENSIONS> cond_dims =
+                    extract_numeric_value_dimensions(
+                        std::move(args[0]), this_->name_, this_->codename_);
+
+                if (!this_->validate_shapes(
+                        extract_numeric_value_dimension(
+                            std::move(args[0]), this_->name_, this_->codename_),
+                        extract_numeric_value_dimension(
+                            std::move(args[1]), this_->name_, this_->codename_),
+                        std::move(cond_dims),
+                        std::move(then_dims)))
+                    HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                        "switch_operation::eval",
+                        util::generate_error_message(
+                            "the condition shape is incompatible with the "
+                            "`then_expression` and `else_expression` "
+                            "shapes",
+                            this_->name_, this_->codename_));
+
+                ir::node_data<std::uint8_t> cond = extract_boolean_value(
+                    std::move(args[0]), this_->name_, this_->codename_);
+                ir::node_data<double> then_expr = extract_numeric_value(
+                    std::move(args[1]), this_->name_, this_->codename_);
+                ir::node_data<double> else_expr = extract_numeric_value(
```

Comment:
Is it ok for switch to always return doubles?

Reply:
yes, I think it always returns doubles